File: v2/snippets/ruby/common/setWorkingDirectoryTests.rb
require 'javonet-ruby-sdk'
require_relative '../../utils/activation_credentials'

RSpec.describe 'Ruby SetWorkingDirectory Tests' do
  it 'Test_SetWorkingDirectory' do
    # <SetWorkingDirectory>
    # define path to be used as Javonet working directory
    user_home_directory = Pathname.new(Dir.home)
    javonet_working_directory = user_home_directory + "myNewDirectory"

    # set Javonet working directory
    Javonet.set_javonet_working_directory(javonet_working_directory.to_s)

    # use activate only once in your app
    Javonet.activate(ActivationCredentials.your_license_key)

    # create other runtime context
    rtm_ctx = Javonet.in_memory.netcore

    # start interact with other runtime...
    # </SetWorkingDirectory>
    response = rtm_ctx.get_type("System.Math").invoke_static_method("Abs", -13).execute
    result = response.get_value
    expect(result).to eq(13)
    expect((javonet_working_directory + "javonet.lic").exist?).to be true
  end
end

File: v2/snippets/javascript/common/setWorkingDirectoryTests.js
const {Javonet} = require('javonet-nodejs-sdk')
const ActivationCredentials = require("../../utils/ActivationCredentials")
const {describe, expect, test} = require("@jest/globals");
const path = require('path');
const fs = require('fs');

describe('Nodejs set working directory tests', () => {

    test(`Test_SetWorkingDirectory`, () => {
        // <SetWorkingDirectory>
        // define path to be used as Javonet working directory
        const userHomeDirectory = require('os').homedir();
        const javonetWorkingDirectory = path.join(userHomeDirectory, "myNewDirectory");

        // set Javonet working directory
        Javonet.setJavonetWorkingDirectory(javonetWorkingDirectory);

        // use Activate only once in your app
        let result = Javonet.activate(ActivationCredentials.yourLicenseKey);
        expect(result).toBe(0);

        // create other runtime context
        let rtmCtx = Javonet.inMemory().netcore();

        // start interact with other runtime...
        // </SetWorkingDirectory>
        let response = rtmCtx.getType("System.Math").invokeStaticMethod("Abs", -13).execute();
        let resultValue = response.getValue();
        expect(resultValue).toBe(13);
        expect(fs.existsSync(path.join(javonetWorkingDirectory, "javonet.lic"))).toBe(true);
    })
})
